[GH-272] restore explicit resource names

davidalpert opened the pull request:
## Description

These explicit resource names were removed in https://github.com/sous-chefs/mongodb/pull/262

The default name of a chef resource, however, is to concatenate
the cookbook name with the resource file name. This cookbook
sets the cookbook name as 'sc-mongodb' which results in a
resource name like 'sc_mongodb_user' or 'sc_mongodb_agent'

These explicit 'provides' calls should allow the cookbook recipes
which use these resources to find them again on Chef 14

### Issues Resolved

- #272 

### Check List
- [x] All tests pass. See https://github.com/sous-chefs/apache2/blob/master/TESTING.md
- [x] ~~New functionality includes testing~~ this change fixes a regression
- [x] ~~New functionality has been documented in the README if applicable~~ this change fixes a regression

Signed-off-by: David Alpert <david@spinthemoose.com>

kitchen-porter commented:
<!--
  1 Error: Please include a CHANGELOG ent...
  1 Warning: This Pull Request is probably ...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please include a CHANGELOG entry.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This Pull Request is probably missing tests.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

ramereth commented:
@davidalpert can you please rebase and resolve the conflicts? I think I fixed most of the issues in #275 but the other suites I commented out may work with this fix so you might enable them.

kitchen-porter commented:
<!--
  0 Errors
  1 Warning: This is a big Pull Request.
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This is a big Pull Request.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

davidalpert commented:
looks like the `mongodb_instance` definition may need to be replaced with a custom resource in order for the specs to pass

davidalpert commented:
but I'm a bit stuck as I cannot run the `delivery` based tests locally since my workstation is pinned to chef 14.x

ramereth commented:
> looks like the `mongodb_instance` definition may need to be replaced with a custom resource in order for the specs to pass

Yeah. I'd revert you uncommenting commit and not worry about the specs until someone converts those definitions to a custom resource.

davidalpert commented:
we've hit #272 again with version `sc-mongodb` v4.1.2 so I'm going to pick this PR up again and try to revive it.

davidalpert commented:
I have rebased onto the latest `upstream/main` and removed both the commenting out of specs and the fix/unfix linting error commits leaving only the original change.

davidalpert commented:
@ramereth I have cleaned up this PR and it seems to pass all the tests 🎉 

would it be possible to merge this and release an update to the cookbook?

ramereth commented on CHANGELOG.md:
```diff
@@ -22,6 +22,7 @@
 - resolved cookstyle error: libraries/user.rb:168:17 convention: `Layout/IndentationConsistency`
 - resolved cookstyle error: libraries/user.rb:169:15 convention: `Layout/IndentationConsistency`
 - resolved cookstyle error: libraries/user.rb:169:17 convention: `Layout/IndentationConsistency`
+- resolved regression in custom resource names (#272)
```
This needs to be moved up to the `## Unreleased` section. Can you please do that?

davidalpert replied:
done

kitchen-porter commented:
Released as: [4.1.3](https://github.com/sous-chefs/sc-mongodb/releases/tag/4.1.3)
